runhouse/main.py

```diff
@@ -455,30 +455,44 @@ def status(
     # case we are inside the cluster
     if cluster_or_local != "file":
         # If we are on the cluster load status directly from the object store
-        cluster_status: dict = dict(obj_store.status())
-        cluster_config = copy.deepcopy(cluster_status.get("cluster_config"))
+        # cluster status type: ResourceStatusData
+        cluster_status, send_status_resp = obj_store.status(send_to_den=True)
+
+        cluster_config = copy.deepcopy(cluster_status.cluster_config)
         current_cluster: Cluster = Cluster.from_config(cluster_config)
-        return _print_status(cluster_status, current_cluster)
 
-    if cluster_name is None:
+    elif cluster_name is None:
```

Comment:
instead of making this an elif, can we make this an if condition first? it feels out of place logically